src/pypromice/process/join_l3.py

```diff
@@ -0,0 +1,282 @@
+    msk = [v for v in var_name.index if v in l3.data_vars]
+    var_name = var_name.loc[msk].to_dict()
+    
+    l3['TA1'] =  l3['TA1'].combine_first(l3['TA3'])
+    l3['TA2'] =  l3['TA2'].combine_first(l3['TA4'])
+    
+    l3=l3.rename(var_name)
+    l3=l3.rename({'timestamp':'time'})
+    return l3
+
+# will be used in the future
+# def aligning_surface_heights(l3m, l3):
+    # df_aux['z_surf_combined'] = \
+    #     df_aux['z_surf_combined'] \
+    #         - df_aux.loc[df_aux.z_surf_combined.last_valid_index(), 'z_surf_combined'] \
+    #          + df_v6.loc[df_v6.z_surf_combined.first_valid_index(), 'z_surf_combined'] 
+    
+    # if s == 'Swiss Camp 10m':        
+    #     df.loc[:df.HS_combined.first_valid_index(), 'HS_combined'] = \
+    #         df2.loc[:df.HS_combined.first_valid_index(), 'HS_combined'] \
+    #             - df2.loc[df2.HS_combined.last_valid_index(), 'HS_combined'] \
+    #              + df.loc[df.HS_combined.first_valid_index(), 'HS_combined'] 
+
+                    
+    # df.loc[df.HS_combined.diff()==0,'HS_combined'] = np.nan
+    
+    # fit = np.polyfit(df.loc[df.HS_combined.notnull(),:].index.astype('int64'),  
+    #                    df.loc[df.HS_combined.notnull(),'HS_combined'], 1)
+    # fit_fn = np.poly1d(fit)
+    
+    # df['HS_combined'] = df['HS_combined'].values \
+    #     - fit_fn(
+    #         df_in.loc[[df_in.z_surf_combined.first_valid_index()],:].index.astype('int64')[0]
+    #     )  +  df_in.loc[df_in.z_surf_combined.first_valid_index(), 'z_surf_combined']
+    # return l3m
+    
+def build_station_dict(config_folder):
+    station_dict = {}
+    
+    # Iterate through all files in the given folder
+    for filename in os.listdir(config_folder):
+        if filename.endswith(".toml"):
+            # Construct full file path
+            file_path = os.path.join(config_folder, filename)
+            
+            # Load TOML file
+            with open(file_path, 'r') as file:
+                data = toml.load(file)
+                station_key = next(iter(data))
+                station_data = data[station_key]
+               
+                if station_data["station_site"] in station_dict:
+                    station_dict[station_data["station_site"]].append(station_key)
+                else:
+                    station_dict[station_data["station_site"]] = [station_key]
+    
+    return station_dict
+    
+def join_l3():
+    args = parse_arguments_joinl3()
+    logging.basicConfig(
+        format="%(asctime)s; %(levelname)s; %(name)s; %(message)s",
+        level=logging.INFO,
+        stream=sys.stdout,
+    )
+    
+    station_dict = build_station_dict(args.config_folder)
+
+    l3m = xr.Dataset()
+    for stid in  station_dict[args.site]:
+        logger.info(stid)
+        
+        is_promice = False
+        is_gcnet = False
+        filepath = os.path.join(args.folder_l3, stid, stid+'_hour.nc')
+        if os.path.isfile(filepath):
```

Comment:
Through one if statement here, we avoid adding a reader attribute to 40+ station config files. I'd say it is leaner this way?

Reply:
You have already read all the station configurations in the function `build_station_list`. I think you should return a list of the station configuration dicts instead of only the station ids. Then you have all the data available at this point.

```
for station_configuration in list_stations:
    stid = station_configurtaion['stid']
    project = station_configurtaion['project']

    is_promice = project == "Promice"
    is_gcnet = project == "GC-Net"
    is_gcnet_historical = project == "historical GC-Net"

```

By the way. Is it correct that the flag `is_gcnet` is referring to the historical dataset?

Reply:
What does "nead"  mean?

Reply:
It stands for Non-Binary Environmental Archive Data (NEAD) format 
https://www.envidat.ch/dataset/nead
it was developed by SLF, WLF and GEUS
basically a CSV with standardized header where attributes an be stored (no info loss compared to netcdf)